src/dmtcp_coordinator.cpp

```diff
@@ -1796,7 +1799,7 @@ int main ( int argc, char** argv )
       setenv(ENV_VAR_CHECKPOINT_DIR, argv[1], 1);
       shift; shift;
     }else if(argc>1 && (s == "-t" || s == "--tmpdir")){
-      setenv(ENV_VAR_TMPDIR, argv[1], 1);
+      tmpDir = Util::calcTmpDir(argv[1]);
```

Comment:
If '--tmpdir' argument is not passed in then, the value of `tmpdir` would be "" and that would be a problem :-). Bottom line is that we need to call calcTmpDir() even if no --tmpdir flag was specified.